src/border_agent/border_agent.cpp

```diff
@@ -143,7 +141,39 @@ BorderAgent::BorderAgent(otbr::Ncp::ControllerOpenThread &aNcp, Mdns::Publisher
     : mNcp(aNcp)
     , mPublisher(aPublisher)
     , mIsEnabled(false)
+    , mVendorName(OTBR_VENDOR_NAME)
+    , mProductName(OTBR_PRODUCT_NAME)
 {
+    setVendorAndProductName(OTBR_VENDOR_NAME, OTBR_PRODUCT_NAME);
+}
+
+void BorderAgent::setVendorAndProductName(const std::string &aVendorName, const std::string &aProductName)
```

Comment:
Should we also trigger `UpdateMeshCopService`?

Reply:
This method is supposed to be called only before Thread is enabled. See the documentation of this API